lib/rules/nesting-selector-no-missing-scoping-root/README.md

```diff
@@ -0,0 +1,66 @@
+# nesting-selector-no-missing-scoping-root
+
+Disallow missing scoping root for nesting selectors.
+
+<!-- prettier-ignore -->
+```css
+    & {}
+/** ↑
+ * This nesting selector */
+```
+
+CSS nesting selectors (`&`) represent the parent selector in nested CSS. When used at the top level or within certain at-rules without a scoping root, they can cause unexpected behavior or indicate a mistake in the CSS structure.
+
+## Options
+
+### `true`
+
+```json
+{
+  "nesting-selector-no-missing-scoping-root": true
+}
+```
+
+The following patterns are considered problems:
+
+<!-- prettier-ignore -->
+```css
+& {}
+```
+
+<!-- prettier-ignore -->
+```css
+@media all {
+  & {}
+}
+```
+
+<!-- prettier-ignore -->
+```css
+@scope (&) {}
+```
+
+The following patterns are _not_ considered problems:
+
+<!-- prettier-ignore -->
+```css
+a {
+  & {}
+}
+```
+
+<!-- prettier-ignore -->
+```css
+a {
+  @media all {
+    & {}
+  }
+}
+```
+
+<!-- prettier-ignore -->
+```css
+a {
+  @scope (&) {}
```

Comment:
off-topic

`at-rule-no-invalid-prelude` has a false negative for this.
ref https://drafts.csswg.org/css-nesting/#example-81212ace